src/js/actions/transform.js

```diff
@@ -1345,10 +1365,10 @@ define(function (require, exports) {
         }
     };
     handleTransformLayer.action = {
-        read: [locks.JS_APP, locks.JS_DOC],
+        read: [locks.JS_APP, locks.JS_DOC, locks.JS_TOOL],
```

Comment:
It seems like this action _writes_ to the tool store. Should it hold the write lock?